src/main/java/org/jboss/forge/arquillian/container/provider/ContainerDirectoryParser.java

```diff
@@ -35,15 +40,23 @@ void createContainers() {
                 new TypeReference<List<Container>>() {
                 });
 
-            containers = Collections.unmodifiableList(parsedContainers);
+            Loader loader = new Loader();
+            final org.arquillian.container.chameleon.spi.model.Container[] containers = loader.loadContainers(Target.class.getClassLoader().getResourceAsStream("chameleon/default/containers.yaml"));
+
+            parsedContainers.stream()
+                .filter(container -> container.getGroupId() == null && container.getArtifactId() == null)
+                .forEach(container -> {
+                    try {
+                        container.setGroupIdAndArtifactIdFromChameleonConfiguration(containers);
```

Comment:
@bartoszmajsak , We are looping over `filtered containers` which don't have groupId & artifactID & they have to go from `setGroupIdAndArtifactIdFromChameleonConfiguration`. I am not sure how can we decouple logic from it?

Am I missing something? Can you please hint me more?